python/tvm/script/builder/tir/var.py

```diff
@@ -15,24 +15,102 @@
 # specific language governing permissions and limitations
 # under the License.
 """TVM Script TIR Buffer"""
-from tvm import tir
-
+from tvm.ir import PrimExpr, Array, Range
+from tvm.tir import Var, IntImm
+from tvm._ffi import register_object as _register_object
+from tvm.runtime import Object, DataType
 from . import _ffi_api
 
 
-class BufferProxy:
-    def __call__(
+@_register_object("script.builder.tir.Buffer")
+class Buffer_(Object):
```

Comment:
just curious - is the class name going to conflict with the `Buffer` variable defined later in this file?

Reply:
yes, it calls itself recursively, until Python crashes